stellar/controllers/stellar_deposit.go

```diff
@@ -0,0 +1,128 @@
+package controllers
+
+import (
+	"database/sql"
+	"net/http"
+	"strings"
+
+	"github.com/stellar/go/clients/horizonclient"
+	"github.com/stellar/go/protocols/horizon/operations"
+
+	"github.com/stellar/go/support/render/problem"
+	"github.com/stellar/starbridge/store"
+)
+
+var (
+	InvalidStellarTxHash = problem.P{
+		Type:   "invalid_stellar_tx_hash",
+		Title:  "Invalid Stellar Transaction Hash",
+		Status: http.StatusBadRequest,
+		Detail: "The transaction hash of the Stellar transaction is invalid.",
+	}
+	StellarTxHashNotFound = problem.P{
+		Type:   "stellar_tx_hash_not_found",
+		Title:  "Stellar Transaction Hash Not Found",
+		Status: http.StatusNotFound,
+		Detail: "The stellar transaction cannot be found.",
+	}
+	StellarTxNotSuccessful = problem.P{
+		Type:   "stellar_tx_not_successful",
+		Title:  "Stellar Transaction Not Successful",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar transaction was not successful.",
+	}
+	StellarTxHasMultipleOperations = problem.P{
+		Type:   "stellar_tx_has_multiple_operations",
+		Title:  "Stellar Transaction Has Multiple Operations",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar deposit transaction is invalid because it has multiple operations.",
+	}
+	StellarTxIsNotPayment = problem.P{
+		Type:   "stellar_tx_is_not_payment",
+		Title:  "Stellar Transaction Is Not Payment",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar deposit transaction is invalid because it is not a payment.",
+	}
+	StellarTxIsNotPaymentToBridge = problem.P{
+		Type:   "stellar_tx_is_not_payment_to_bridge",
+		Title:  "Stellar Transaction Is Not Payment To Bridge",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The stellar deposit transaction is invalid because it " +
+			"is not a payment to the bridge account.",
+	}
+	InvalidEthereumRecipient = problem.P{
+		Type:   "invalid_ethereum_recipient",
+		Title:  "Invalid Ethereum Recipient",
+		Status: http.StatusUnprocessableEntity,
+		Detail: "The recipient of the deposit is not a valid Ethereum address.",
+	}
+	// TODO: remove this once getStellarDeposit is used
+	_ = getStellarDeposit
+)
+
+func getStellarDeposit(client *horizonclient.Client, bridgeAccount string, depositStore *store.DB, r *http.Request) (store.StellarDeposit, error) {
+	txHash := strings.TrimPrefix(r.PostFormValue("transaction_hash"), "0x")
+	if !validTxHash.MatchString(txHash) {
+		return store.StellarDeposit{}, InvalidStellarTxHash
+	}
+
+	deposit, err := depositStore.GetStellarDeposit(r.Context(), txHash)
+	if err == nil {
+		return deposit, nil
+	} else if err != sql.ErrNoRows {
+		return store.StellarDeposit{}, err
+	}
+
+	ops, err := client.Operations(horizonclient.OperationRequest{
```

Comment:
I think it can be a problem for Horizons with partial history. Maybe we should store the incoming payments in a DB (as we do for withdrawals in `stellar.Observer`)? I guess we'd need to introduce a minimum deposit amount to prevent spamming the account with stroop payments.

Reply:
I think that makes sense since we are already processing incoming payments